pkg/autoupdate/tuf/library_lookup.go

```diff
@@ -4,21 +4,124 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"os"
 	"path/filepath"
+	"runtime"
+	"strings"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/kolide/launcher/pkg/autoupdate"
+	"github.com/peterbourgon/ff/v3"
 )
 
 type BinaryUpdateInfo struct {
 	Path    string
 	Version string
 }
 
+type autoupdateConfig struct {
+	rootDirectory        string
+	updateDirectory      string
+	channel              string
+	localDevelopmentPath string
+}
+
+var channelsUsingLegacyAutoupdate = map[string]bool{
+	"stable": true,
+	"beta":   true,
+}
+
+// CheckOutLatestWithoutConfig returns information about the latest downloaded executable for our binary,
+// searching for launcher configuration values in its config file.
+// For now, it is only available when launcher is on the nightly update channel.
+func CheckOutLatestWithoutConfig(binary autoupdatableBinary, logger log.Logger) (*BinaryUpdateInfo, error) {
+	cfg, err := getAutoupdateConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not get autoupdate config: %w", err)
+	}
+
+	// Short-circuit lookup for local launcher test builds
+	if binary == binaryLauncher && cfg.localDevelopmentPath != "" {
+		return &BinaryUpdateInfo{Path: cfg.localDevelopmentPath}, nil
+	}
+
+	return CheckOutLatest(binary, cfg.rootDirectory, cfg.updateDirectory, cfg.channel, logger)
+}
+
+// getAutoupdateConfig reads launcher's config file to determine the configuration values
+// needed to work with the autoupdate library.
+func getAutoupdateConfig() (*autoupdateConfig, error) {
+	configFilePath := getConfigFilePath(os.Args[1:])
+	if configFilePath == "" {
+		return nil, errors.New("could not get config file path")
+	}
+	if _, err := os.Stat(configFilePath); err != nil && os.IsNotExist(err) {
+		return nil, fmt.Errorf("could not read config file because it does not exist at %s: %w", configFilePath, err)
+	}
+
+	cfgFileHandle, err := os.Open(configFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not open config file %s for reading: %w", configFilePath, err)
+	}
+	defer cfgFileHandle.Close()
+
+	cfg := &autoupdateConfig{}
+	if err := ff.PlainParser(cfgFileHandle, func(name, value string) error {
+		switch name {
+		case "root_directory":
+			cfg.rootDirectory = value
+		case "update_directory":
+			cfg.updateDirectory = value
+		case "update_channel":
+			cfg.channel = value
+		case "localdev_path":
+			cfg.localDevelopmentPath = value
+		}
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("could not parse config file %s: %w", configFilePath, err)
+	}
+
+	return cfg, nil
+}
+
+// getConfigFilePath returns the path to launcher's launcher.flags file. If the path
+// is available in the command-line args, it will return that path; otherwise, it
+// will fall back to a well-known location.
+func getConfigFilePath(args []string) string {
```

Comment:
might be good to expose this as a public function in `/pkg/launcher`, currently working on a flare for config and this would be handy

Reply:
Yeah I was thinking about that as I was reading your PR! I'll move it